File: cs61a/mutability/withdraw.js
function makeWithdrawList(balance) {
    let my_balance = balance

    return function withdraw(amount) {
        if (amount > my_balance) {
            return "Insufficient Balance!"
        }
        my_balance -= amount
        return my_balance
    }
}

withdraw = makeWithdrawList(100)
console.log(withdraw(25))
console.log(withdraw(25))
console.log(withdraw(25))
console.log(withdraw(25))
console.log(withdraw(25))

File: cs61a/mutability/withdraw.py
def make_withdraw_list(balance: int = 100):
    my_balance = balance

    def withdraw(amount: int):
        nonlocal my_balance
        my_balance -= amount
        return my_balance

    return withdraw

def make_withdraw_list_2(balance: int = 100):
    my_balance = [balance]

    def withdraw(amount: int):
        if amount > my_balance[0]:
            return "insufficient balance"
        
        my_balance[0] -= amount
        return my_balance[0]

    return withdraw
